### PR TITLE
Fix path duplicate functions and add value hash/equal benchmark

### PR DESCRIPTION
**Pull request by yixinglu**
benchmark between string and value.

### CODE REVIEW

**Review by nevermore3 — DISMISSED** [review]

**Review by nevermore3 — APPROVED** [review]

File: src/common/datatypes/Path.cpp [hunk 1]
```diff
@@ -57,20 +57,24 @@ bool Path::hasDuplicateEdges() const {
...
+        const auto& edgeSrc = step.type > 0 ? srcVid : dstVid;
+        const auto& edgeDst = step.type > 0 ? dstVid : srcVid;
+        auto edgeKey = folly::stringPrintf("%s%ld%s%ld%s%ld%ld",
+                                           edgeSrc.c_str(),
+                                           edgeSrc.size(),
+                                           edgeDst.c_str(),
+                                           edgeDst.size(),
                                            step.name.c_str(),
+                                           step.name.size(),
                                            step.ranking);
```
Comment (CPWstatic) [inline src/common/datatypes/Path.cpp:72]: Would it be faster by using tuple?

Comment (yixinglu) [reply src/common/datatypes/Path.cpp]: Yeah. Fixed.

File: src/common/datatypes/Path.cpp [hunk 2]
```diff
@@ -56,21 +57,18 @@ bool Path::hasDuplicateEdges() const {
     if (steps.size() < 2) {
         return false;
     }
-    std::unordered_set<std::string> uniqueSet;
+    using Key = std::tuple<Value, Value, std::string, EdgeRanking>;
```
Comment (Shylock-Hg) [inline src/common/datatypes/Path.cpp:60]: I think you could use edgeType instead of edge name.

Comment (yixinglu) [reply src/common/datatypes/Path.cpp]: OK, fixed.

**Review by Aiee — APPROVED** [review]

File: src/common/datatypes/Path.cpp [hunk 3]
```diff
@@ -56,21 +57,18 @@ bool Path::hasDuplicateEdges() const {
...
-        auto edgeDst = step.type > 0 ? step.dst.vid : srcVid;
-        auto edgeKey = folly::stringPrintf("%s%s%s%ld",
-                                           edgeSrc.toString().c_str(),
-                                           edgeDst.toString().c_str(),
-                                           step.name.c_str(),
-                                           step.ranking);
-        auto res = uniqueSet.emplace(std::move(edgeKey));
-        if (!res.second) {
+        const auto& dstVid = step.dst.vid;
+        const auto& edgeSrc = step.type > 0 ? srcVid : dstVid;
```
Comment (laura-ding) [inline src/common/datatypes/Path.cpp:65]: It's not needed here anymore

**Review by laura-ding — APPROVED** [review]

**Pull request merged**